lib/teleterm/services/connectmycomputer/connectmycomputer.go

```diff
@@ -502,10 +502,16 @@ type NodeDelete struct {
 // Run grabs the host UUID of an agent from a disk and deletes the node with that name.
 func (n *NodeDelete) Run(ctx context.Context, presence Presence, cluster *clusters.Cluster) error {
 	hostUUID, err := utils.ReadHostUUID(getAgentDataDir(n.cfg.AgentsDir, cluster.ProfileName))
+	if trace.IsNotFound(err) {
+		return nil
```

Comment:
Not a blocker, but IMHO `NodeDelete` should not ignore this error. If we get a request to delete a node but the data dir is not found, we should error as at this point we're unable to tell if the node exists or not.

Instead, `removeAgent` in `connectMyComputerContext` should first call `connectMyComputerService.isAgentConfigFileCreated(rootClusterUri)` before attempting to call `ctx.connectMyComputerService.removeConnectMyComputerNode`.

Reply:
I'm not sure if calling `isAgentConfigFileCreated(rootClusterUri)` before `removeConnectMyComputerNode` gives us anything. The only situation where you can click "Remove agent" is on the status page which is visible only when the config file is created, so the check for `isAgentConfigFileCreated` will probably always return `true`.

The reason for ignoring this NOT_FOUND error is that it allows us to remove the agent when the configuration file was created, but the agent did not start for some reason (so there is no data directory).
To keep this behaviour, we could call `getConnectMyComputerNodeName` before `removeConnectMyComputerNode and check if it returns NOT_FOUND. 
However, I don't know if this is better than just ignoring the error on the Go side.

Let me know what you think about it :)

Reply:
Ah, you're right, I conflated checking the existence of the config file with checking the existence of the data folder (and thus the node UUID).

This nit was mostly just from the perspective of an overall approach to building software, not this specific case. If in the future something needs to consider lack of the data dir to be a show-stopping error, there'll be no choice but to refactor this as `NodeDelete` already made the assumption that the lack of data dir is something that can be ignored.

It can stay like this I think since something else needing to delete the node and having a different approach to error handling is rather unlikely. 🤞